core/store/src/columns.rs

```diff
@@ -327,6 +350,72 @@ impl DBCol {
             _ => false,
         }
     }
+
+    /// Whether this column should be copied to the cold storage.
+    pub const fn is_cold(&self) -> bool {
+        match self {
+            DBCol::Block => true,
+            _ => false,
+        }
+    }
+
+    /// Vector of DBKeyType s concatenation of which results in key for the column.
+    pub fn key_type(&self) -> Vec<DBKeyType> {
```

Comment:
This could have `&'static [DBKeyType]` return type.  No need to allocate new vector each time.